aws/scenarios/vm/ec2VM/ec2VM.go

```diff
@@ -11,23 +11,45 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 )
 
+type EC2VM struct {
+	env aws.Environment
+	commonvm.VM
+}
+
 // NewEc2VM creates a new EC2 instance. By default use WithOS(os.UbuntuOS, os.AMD64Arch).
-func NewEc2VM(ctx *pulumi.Context, options ...func(*Params) error) (commonvm.VM, error) {
+func NewEc2VM(ctx *pulumi.Context, options ...func(*Params) error) (*EC2VM, error) {
 	return newVM(ctx, options...)
 }
 
+func (vm *EC2VM) GetAwsEnvironment() aws.Environment {
+	return vm.env
+}
+
+type EC2UnixVM struct {
+	*commonvm.UnixVM
+	*EC2VM
```

Comment:
`*EC2VM` includes `commonvm.VM` which included by `commonvm.UnixVM`. Maybe that is the reason why you have both  a non pointer type (`commonvm.VM`) and a pointer type (`*commonvm.UnixVM`).
Instead, you can define a new struct that has the method `GetAwsEnvironment()` and embdeded it in both `EC2VM` and `EC2UnixVM`.
```go
type Infra struct { env aws.Environment }

type EC2VM struct {
	Infra
	*commonvm.VM
}

type EC2UnixVM struct {
	Infra
	*commonvm.VM
}

```